src/components/SpruceForm/index.tsx

```diff
@@ -13,31 +14,37 @@ import widgets from "./Widgets";
 
 export const SpruceForm: React.VFC<SpruceFormProps> = ({
   customFormatFields,
+  customValidate,
   disabled,
   fields,
   formData,
   onChange,
   schema,
   tagName,
   uiSchema,
-  validate,
 }) => (
   <Form
-    fields={{ ...baseFields, ...fields }}
+    fields={fields}
     schema={schema}
     onChange={onChange}
     widgets={widgets}
```

Comment:
There are some new types that can be applied to widgets. https://rjsf-team.github.io/react-jsonschema-form/docs/advanced-customization/custom-widgets-fields/#custom-component-registration